sdk/src/quotes/public/swap-quote.ts

```diff
@@ -45,6 +46,62 @@ export type SwapQuote = {
   estimatedAmountOut: u64;
 } & SwapInput;
 
+export async function swapQuoteByInputToken(
+  whirlpool: Whirlpool,
+  swapTokenMint: Address,
+  tokenAmount: u64,
+  amountSpecifiedIsInput: boolean,
+  slippageTolerance: Percentage,
+  fetcher: AccountFetcher,
+  programId: Address
+): Promise<SwapQuote> {
+  const whirlpoolData = whirlpool.getData();
+  const swapMintKey = AddressUtil.toPubKey(swapTokenMint);
+  invariant(
+    whirlpoolData.tokenMintA.equals(swapMintKey) || whirlpoolData.tokenMintB.equals(swapMintKey),
+    "swapTokenMint does not match any tokens on this pool"
+  );
+
+  const aToB =
+    swapMintKey.equals(whirlpoolData.tokenMintA) === amountSpecifiedIsInput ? true : false;
+
+  const tickArrayAddresses = PoolUtil.getTickArrayPublicKeysForSwap(
+    whirlpoolData.tickCurrentIndex,
+    whirlpoolData.tickSpacing,
+    aToB,
+    AddressUtil.toPubKey(programId),
+    whirlpool.getAddress()
+  );
+  const tickArrays = await fetcher.listTickArrays(tickArrayAddresses, true);
```

Comment:
if this is the publicly exposed method, it might be good to have the refresh here be optional?